skyportal/handlers/api/observingrun.py

```diff
@@ -0,0 +1,234 @@
+from sqlalchemy.orm import joinedload
+from marshmallow.exceptions import ValidationError
+from baselayer.app.access import permissions, auth_or_token
+from ..base import BaseHandler
+from ...models import(DBSession, ObservingRun, ClassicalAssignment, Obj,
+                      Thumbnail)
+from ...schema import (ObservingRunPost, ObservingRunGet,
+                       ObservingRunGetWithAssignments)
+
+
+class ObservingRunHandler(BaseHandler):
+
+    @permissions(['Upload data'])
+    def post(self):
+        """
+        ---
+        description: Add a new observing run
+        requestBody:
+          content:
+            application/json:
+              schema: ObservingRunPost
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            id:
+                              type: integer
+                              description: New Observing Run ID
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        # See bottom of this file for redoc docstring -- moved it there so that
+        # it could be made an f-string.
+
+        data = self.get_json()
+
+        try:
+            rund = ObservingRunPost.load(data)
+        except ValidationError as exc:
+            return self.error('Invalid/missing parameters: '
+                              f'{exc.normalized_messages()}')
+
+        run = ObservingRun(**rund)
+
+        current_user_id = self.current_user.id if \
```

Comment:
this whole thing can be simplified to `self.associated_user_object.id`

Reply:
(admittedly this was probably copied from another handler that needs updating, too)

Reply:
Fixed